src/osd/OSDMap.cc

```diff
@@ -1769,6 +1769,18 @@ void OSDMap::maybe_remove_pg_upmaps(CephContext *cct,
     vector<int> raw;
     int primary;
     tmpmap.pg_to_raw_up(pg, &raw, &primary);
+    auto size = tmpmap.get_pg_pool_size(pg);
```

Comment:
the reason why `tmpmap.get_pg_pool_size(pg)` or `tmpmap.get_pg_pool_crush_rule(pg)` fails is that the pg does not exist anymore. does it make sense to check `tmpmap.pg_exists(pg)` before moving further?